docs/src/main/sphinx/develop/event-listener.rst

```diff
@@ -9,9 +9,12 @@ events:
 * Query completion (success or failure)
 * Split completion (success or failure)
 
+Event details include session, query execution, resource utilization, timeline,
```

Comment:
Since the event listener is actually developer facing documentation - how do we feel about linking each of the events above to their corresponding files in code?

cc: @electrum

Reply:
Such links would be stable until the next refactor. How much risk is there of broken links in the future?

Reply:
The classes won't get renamed most likely because it's part of the SPI and public API. I can't remember any time where public API was broken except the project renames.

I'd like to get @electrum's views on this too.

Reply:
I would vote for voting to the relevant SPI docs instead. And if we really think that is needed, then link to the code from there (and that would be tbd in separate PR)

Reply:
The SPI docs on the event listener classes (how the event looks like) doesn't exist (and I don't think documenting all the fields makes sense).

I was thinking about simply linking each of the 3 "Query completion"/"Split completion"/"Query created" bullets to the respective classes in the SPI:

- [Query Completed](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/eventlistener/QueryCompletedEvent.java)
- [Query Created](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/eventlistener/QueryCreatedEvent.java)
- [Split Completed](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/eventlistener/SplitCompletedEvent.java)

Reply:
I think this belongs to the SPI docs nevertheless. Maybe we should link directly in the meantime .. wdyt @electrum ?